tests/testthat/test-register_logger.R

```diff
@@ -14,7 +14,13 @@ testthat::test_that("register_logger throws an error when passed an invalid name
 
 testthat::test_that("register_logger throws an error when setting the logger layout fails", {
   testthat::expect_error(
-    withr::with_options(new = list(teal.log_layout = stop), code = register_logger("test")),
+    withr::with_envvar(
+      new = list(TEAL.LOG_LAYOUT = ""),
+      code = withr::with_options(
+        new = list(teal.log_layout = stop),
+        code = register_logger("test")
+      )
+    ),
```

Comment:
Yes, this is fine. 

Related with `get_val` function to pull both environment variable and option names.
https://github.com/insightsengineering/teal.logger/blob/1e34763b71442fc4f12627813aff4017143a4f6a/R/utils.R#L37-L39